### PR TITLE
Fixing 1295535 via doing action control index handover between keys.

### PR DESCRIPTION
**Pull request by jimon**

### CODE REVIEW

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ however, it has to be formatted properly to pass verification tests.
   * In our own measurements, `InputUser.OnEvent` is >9 times faster than before and `RebindingOperation.OnEvent` is ~2.5 times faster.
 - Fixed PS4 controller not recognized on Mac when connected over Bluetooth ([case 1286449](https://issuetracker.unity3d.com/issues/input-system-dualshock-4-zct1e-dualshock-2-v1-devices-are-not-fully-recognised-over-bluetooth)).
 - Fixed restart prompt after package installation not appearing on Unity 2020.2+ ([case 1292513](https://issuetracker.unity3d.com/issues/input-system-after-package-install-the-update-slash-switch-and-restart-prompt-does-not-appear)).
+- Fixed input system not registering multiple inputs ([case 1295535](https://issuetracker.unity3d.com/issues/input-system-not-registering-multiple-inputs)).
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/CHANGELOG.md]:
```
- Fixed action with multiple bindings getting stuck in `Performed` state when two or more controls are pressed at the same time ([case ...](...).
  * Regression introduced in 1.1-preview.2.
```

CHANGELOG entries should try to make it as easy as possible for the user to figure out whether a change is relevant or not.

File: Assets/Tests/InputSystem/CoreTests_Actions.cs [hunk 2]
```diff
@@ -8388,4 +8388,31 @@ public void TODO_Actions_ReResolvingBindings_DoesNotAllocate_IfXXX()
...
+    public void Actions_Buttons_CanTransitionToDifferentActuationControl_WhilePerformed()
+    {
+        var keyboard = InputSystem.AddDevice<Keyboard>();
+        var wasCanceled = false;
+        var action = new InputAction("action", InputActionType.Button);
+        action.canceled += _ => wasCanceled = true;
+        action.AddBinding("<Keyboard>/s");
+        action.AddBinding("<Keyboard>/a");
+        action.Enable();
+        wasCanceled = false;
```
Comment (Rene-Damm) [inline Assets/Tests/InputSystem/CoreTests_Actions.cs]: Dead code.

File: Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs [hunk 3]
```diff
@@ -1451,6 +1451,13 @@ private void ProcessDefaultInteraction(ref TriggerState trigger, int actionIndex
                         var threshold = pressPoint * ButtonControl.s_GlobalDefaultButtonReleaseThreshold;
                         if (actuation <= threshold)
                             ChangePhaseOfAction(InputActionPhase.Canceled, ref trigger);
+                        else
+                        {
+                            // Always remember the latest trigger control index
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs]: Suggest
```
// ShouldIgnoreControlStateChange may have switched one from control to another,
// so make sure we update the trigger state here regardless of whether we changed
// phase or not.
```

File: Assets/Tests/InputSystem/CoreTests_Actions.cs [hunk 4]
```diff
@@ -8388,4 +8388,31 @@ public void TODO_Actions_ReResolvingBindings_DoesNotAllocate_IfXXX()
     {
         Assert.Fail();
     }
+
+    [Test]
+    [Category("Actions")]
+    public void Actions_Buttons_CanTransitionToDifferentActuationControl_WhilePerformed()
```
Comment (Rene-Damm) [inline Assets/Tests/InputSystem/CoreTests_Actions.cs]: Maybe `Actions_OnActionWithMultipleBindings_CanTransitionFromOneActuatedControlToAnother` (there's a couple OnActionWithMultipleBindings tests already)

**Review by Rene-Damm — CHANGES_REQUESTED** [review]

File: Assets/Tests/InputSystem/CoreTests_Actions.cs (hunk 4 above)
Comment (jimon) [reply Assets/Tests/InputSystem/CoreTests_Actions.cs]: Fixed, moved the test closer to them

File: Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs (hunk 3 above)
Comment (jimon) [reply Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs]: Fixed

File: Assets/Tests/InputSystem/CoreTests_Actions.cs (hunk 2 above)
Comment (jimon) [reply Assets/Tests/InputSystem/CoreTests_Actions.cs]: Removed

File: Packages/com.unity.inputsystem/CHANGELOG.md (hunk 1 above)
Comment (jimon) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: Ok, fixed

**Pull request merged**